### ISSUE TITLE
Allow for QR code download

### ISSUE THREAD

**Issue by andrewtavis** [activist-org/activist#472]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

Organization and event pages are getting a QR code component in #353, with #470 being the contribution to add this feature. The next step for this will be to allow an SVG of the QR code to be downloaded by the user so that I can be added to real life media.

Note that https://github.com/scopewu/qrcode.vue/issues/50#issuecomment-1055585303 has some good ideas for how to implement download functionality for `qrcode.vue` (see other comments in the issue as well). The general goal would be to have this first download an SVG, with other download options being covered by another issue.

- Please note that this issue is blocked by #353

### Contribution

I've already discussed this issue with @VinnyHS2 (please write in the issue so I can assign) :) Happy to support! 😊

**Comment by VinnyHS2** [comment]
I would love to continue this work, if it is okay

**Comment by andrewtavis** [comment]
You're the exact person to work on this, @VinnyHS2 😊 Assigning now :)

**Comment by VinnyHS2** [comment]
@andrewtavis I tested the solution you sent, but it only exports the QRCode, not the whole thing. If you just want the QRCode the solution will work, but I guess that is not what you expect.

**Comment by andrewtavis** [comment]
Hey @VinnyHS2 :) Thanks for the tests so far! I guess that exporting the full div would be the ideality here. Do you have some ideas on this?

**Comment by VinnyHS2** [comment]
Well... I have to test, but I have saw some libs to do this. I keep you update about it.

**Comment by andrewtavis** [comment]
Thanks, @VinnyHS2!

**Comment by wkyoshida** [comment]
Could an idea be to separate out the QRCode div into a separate component out from [`ModalQRCode.vue`](https://github.com/activist-org/activist/blob/main/frontend/components/modal/ModalQRCode.vue)? We could potentially then:
- use this new component in `ModalQRCode.vue` itself
- export just the QR code itself as we are able to now and then reconstruct the nice activist-branded one using this new component

**Comment by andrewtavis** [comment]
I tried to thumbs up this comment twice, @wkyoshida 😅 How does this sound, @VinnyHS2?

**Comment by VinnyHS2** [comment]
Sounds really good for me, I like the idea

**Comment by VinnyHS2** [comment]
@andrewtavis I got a little confused, so now i'm supposed to export only the qrcode image? and then in another moment, we will create this new component to export everything?

**Comment by andrewtavis** [comment]
I'd say it might be best to jump straight to making the component for the QR code, and then we work on exporting that. How does that sound, @VinnyHS2?

**Comment by VinnyHS2** [comment]
Sounds great, I was just confused hahaha. I'm trying to use the lib html2canvas to do it, but I'm struggling to make it work, just like the problem i had before

**Comment by andrewtavis** [comment]
As far as importing it in? Things to try:

- Using Yarn to install it
- `docker-compose up --build` afterwards
- `docker-compose down` and/or deleting the container on Docker Desktop and restarting from scratch
- Deleting the `.nuxt` directory

Let us know if it's still not working and we can trouble shoot some more! 😊

**Comment by VinnyHS2** [comment]
It worked. I don't know why, because I already tryied hahahaha

**Comment by andrewtavis** [comment]
Hey @VinnyHS2 👋 72af0f6 sadly needed to revert the download because I was having lots of problems with the running Docker once I brought it in... It does send through the formatting that was needed and the icon is now fixed. 

Specifically what I was seeing with the new dependencies was `[plugin:vite:import-analysis] Missing "./dist/runtime/composables" specifier in "@nuxtjs/i18n" package`. Is likely something with an update to one screwing up another, and I did try to do a full update to the frontend dependencies via `yarn upgrade`, but it just wasn't working.

Let me think on this a bit and we can try to get back to it! Let me know if you have any thoughts 😊 As of now the part that needs `html2canvas` has just been commented out :)

**Comment by andrewtavis** [comment]
Ok then @VinnyHS2! 😊 Now both of us have had random problems being solved by doing the same thing again in this issue 😅 All's good now! I made a few more changes:

- The background is transparent so we're getting just the code and the boarder around it and not a white rectangle
- There's now a bit of a shadow under it in the modal, but this is not included in the download
- I made a variable for the name of the organization or event that's then passed to the QR code so that we can use it in the file name

The commit also includes a few more minor changes for other issues or random stuff I'd had in my notes to fix :)

Thanks for the hard work to get this done! Do you have interest in moving on to #473?

**Comment by VinnyHS2** [comment]
I just saw it, you had some of the problems that I was having hahaha. Noticed that someone just got the other issue :(
I will be looking for new issues

**Comment by andrewtavis** [comment]
Ya they messaged me and said they'd already started on it, so because of that I figure it'd be ok to let them work on it. I'll be making some more tonight/tomorrow at the latest!
